### DISCUSSION TITLE
Strange behaviour when using removeValue mode="array"

### DISCUSSION THREAD

**Discussion by nbentoneves** [TanStack/form#1000] [Q&A] [answered]
I'm having an issue with the `removeValue` function from `FieldApi`. I've created an example based on https://tanstack.com/form/latest/docs/framework/react/guides/arrays#full-example.

You can reproduce the problem by following these steps:

1. Add a person with Name: A and Age: 10.
2. Add a person with Name: B and Age: empty.
3. Remove the person with Name: A.

Currently, the result is:
- Name: B and Age: 10

However, the expected result is:
- Name: B and Age: empty.

I’m not sure if I'm doing something wrong. Could someone please take a look at the code and see if this might be a bug? 

Thank you! https://codesandbox.io/p/devbox/crazy-wind-64cdcd?embed=1

**Comment by Balastrong** [comment, marked as answer]
I think it's because you're having both controlled and uncontrolled inputs (with value undefined).

Does it happen the same with this? `value={subField.state.value ?? ''}`

**Reply by nbentoneves** [reply]
> Does it happen the same with this? value={subField.state.value ?? ''}

No, with `?? ""` I can get the expected behaviour. 🥸
